google/cloud/security/enforcer/gce_firewall_enforcer.py

```diff
@@ -486,6 +486,9 @@ def add_rules(self, rules, network_name=None):
           InvalidFirewallRuleError: One or more rules failed validation.
         """
         for rule in rules:
```

Comment:
This is the correct check to make sure it is a valid dictionary.  So, LGTM, and addressing andrew's comment.